torch/quantization/fx/quantize.py

```diff
@@ -958,15 +986,19 @@ def _find_quants(self, graph: Graph, matches: Dict[str, MatchResult],
 
         def visit(node, matched_pattern, qconfig):
             def visit_arg(arg):
-                is_weight = False
-                if isinstance(node, Node) and node.op == 'call_function' and \
-                        node.target in WEIGHT_INDEX_DICT:
-                    for i, node_arg in enumerate(node.args):
-                        if arg is node_arg and i in \
-                                WEIGHT_INDEX_DICT[node.target]:  # type: ignore
-                            is_weight = True
-                if qconfig is not None and \
-                   (activation_is_statically_quantized(qconfig) or is_weight):
+                is_weight = node_arg_is_weight(node, arg)
+                is_bias = node_arg_is_bias(node, arg)
+                should_add_handler = (
+                    # qconfig must be set to observe
+                    qconfig is not None and (
+                        # either activations or weights can be observed
+                        activation_is_statically_quantized(qconfig) or
+                        is_weight
+                    ) and
+                    # bias should not be observed
+                    (not is_bias)
```

Comment:
I feel the condition should probably be:
```
qconfig is not None and ((activation_is_statically_quantized(qconfig) and not is_bias) or (activation_is_statically_quantized(qconfig) and is_weight)
```

# we didn't check the weight should be statically quantized before actually

for this PR I think we can merge (not is_bias) check with activation_is_statically_quantized check and add the weight check in a separate PR.

Basically
1. `(activation_is_statically_quantized(qconfig) and not is_bias` checks if the activation needs to be observed and we'll use qconfig.activation as act_post_process_ctr
2. `(activation_is_statically_quantized(qconfig) and is_weight)` checks if the weight needs to be observed and we'll use qconfig.weight as act_post_process_ctr

Reply:
I agree, what I have now is not the most readable.  Thinking about it some more, activation / weight / bias seem mutually exclusive, so maybe it should be 

```
is_activation = not (is_weight or is_bias)
(qconfig is not None and (
  (is_activation and activation_is_statically_quantized(qconfig)) or
  (is_weight and weight_is_statically_quantized(qconfig))
)
```
?

Reply:
sure, this sounds good